src/linesearch.jl

```diff
@@ -115,8 +115,12 @@ function perform_line_search(
     workspace,
     memory_mode,
 )
+    dd = fast_dot(d, d)
+    if dd == 0
+        return 0
+    end
```

Comment:
```suggestion
    if dd <= eps(eltype(dd))
        return dd
    end
```
that way we ensure type stability